packages/animations/example/lib/shared_axis_transition/shared_axis_transition.dart

```diff
@@ -0,0 +1,252 @@
+                    RaisedButton(
+                      onPressed: isLoggedIn ? null : toggleLoginStatus,
+                      color: Theme.of(context).colorScheme.primary,
+                      textColor: Theme.of(context).colorScheme.onPrimary,
+                      disabledColor: Colors.black12,
+                      child: const Text('NEXT'),
+                    ),
+                  ],
+                ),
+              ),
+              const Divider(thickness: 2.0),
+              Row(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: <Widget>[
+                  Radio<SharedAxisTransitionType>(
+                    value: SharedAxisTransitionType.horizontal,
+                    groupValue: transitionType,
+                    onChanged: (SharedAxisTransitionType newValue) {
+                      updateTransitionType(newValue);
+                    },
+                  ),
+                  const Text('X'),
+                  Radio<SharedAxisTransitionType>(
+                    value: SharedAxisTransitionType.vertical,
+                    groupValue: transitionType,
+                    onChanged: (SharedAxisTransitionType newValue) {
+                      updateTransitionType(newValue);
+                    },
+                  ),
+                  const Text('Y'),
+                  Radio<SharedAxisTransitionType>(
+                    value: SharedAxisTransitionType.scaled,
+                    groupValue: transitionType,
+                    onChanged: (SharedAxisTransitionType newValue) {
+                      updateTransitionType(newValue);
+                    },
+                  ),
+                  const Text('Z'),
+                ],
+              ),
+            ],
+          );
+        },
+      ),
+    );
+  }
+}
+
+class _CoursePage extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: <Widget>[
+        const Padding(padding: EdgeInsets.symmetric(vertical: 8.0)),
+        Text(
+          'Streamling your courses',
+          style: Theme.of(context).textTheme.headline,
+        ),
+        const Padding(padding: EdgeInsets.symmetric(vertical: 5.0)),
+        Text(
+          'Bundled categories appear as groups in your feed.'
+          'You can always change this later',
+          style: TextStyle(
+            fontSize: 12.0,
+            color: Colors.grey,
+          ),
+          textAlign: TextAlign.center,
+        ),
+        const _CourseSwitch(course: 'Arts & Crafts'),
+        const _CourseSwitch(course: 'Business'),
+        const _CourseSwitch(course: 'Illustration'),
+        const _CourseSwitch(course: 'Design'),
+        const _CourseSwitch(course: 'Culinary'),
+      ],
+    );
+  }
+}
+
+class _CourseSwitch extends StatefulWidget {
+  const _CourseSwitch({
+    this.course,
+  });
+
+  final String course;
+
+  @override
+  __CourseSwitchState createState() => __CourseSwitchState();
```

Comment:
nit: personally, I am not a fan of the "double" private. One underscore is enough.

Reply:
This was an accident -- I think using the VSCode refactor shortcut for `StatelessWidget` -> `StatefulWidget` automatically adds the underscore since my StatelessWidget started off private. I'll remove the additional underscores